kafka-rest/src/main/java/io/confluent/kafkarest/Utils.java

```diff
@@ -43,4 +53,44 @@ public static int errorCodeFromProducerException(Throwable e) {
     }
   }
 
+  public static RestException convertProducerException(KafkaException exception) {
+    if (StringUtil.startsWithIgnoreCase(exception.getMessage(), "Invalid partition")) {
+      throw Errors.partitionNotFoundException();
```

Comment:
Should we be consistent and just return the error here? Same as in the `throw Errors.kafkaErrorException(exception);` line